src/ScalarAuthClient.js

```diff
@@ -94,11 +95,43 @@ class ScalarAuthClient {
         });
     }
 
+    _checkToken(token) {
+        return this._getAccountName(token).then(userId => {
+            const me = MatrixClientPeg.get().getUserId();
+            if (userId !== me) {
+                throw new Error("Scalar token is owned by someone else: " + me);
+            }
+            return token;
+        }).catch((e) => {
+            if (e instanceof TermsNotSignedError) {
+                console.log("Integrations manager requires new terms to be agreed to");
+                // The terms endpoints are new and so live on standard _matrix prefixes,
+                // but IM rest urls are currently configured with paths, so remove the
+                // path from the base URL before passing it to the js-sdk
```

Comment:
Could you copy the note about a grace period from https://github.com/matrix-org/matrix-react-sdk/commit/7d7878245bb93789ece48806cbc43f2edb9480e5 into the code here as well? I am worried we'll forget about it otherwise.